### PR TITLE
Feature/school 464/change user creation

### PR DESCRIPTION
**Pull request by Feazom**

### CODE REVIEW

**Review by zavarin-michael — COMMENTED** [review]
we should think about it more

File: open_schools_platform/common/models.py [hunk 1]
```diff
@@ -1,17 +1,64 @@
...
+        Regular update_or_create() fails on soft-deleted, existing record with unique constraint on non-id field
+        If object is soft-deleted we don't update-or-create it but reset the deleted field to None.
+        So the object is visible again like a create in any other case.
+
+        Attention: If the object is "revived" from a soft-deleted state the created return value will
+        still be false because the object is technically not created unless you set
+        SAFE_DELETE_INTERPRET_UNDELETED_OBJECTS_AS_CREATED = True in the django settings.
+
+        Args:
+            defaults: Dict with defaults to update/create model instance with
```
[the remark this answers is not in the document]
Comment (Feazom) [reply open_schools_platform/common/models.py:27]: yes

File: open_schools_platform/organization_management/employees/models.py [hunk 2]
```diff
@@ -25,23 +25,16 @@ def create(self, *args: Any, **kwargs: Any):
...
-        employee_profile = self.model(
-            *args,
-            **kwargs,
-        )
-
-        employee_profile.full_clean()
-        employee_profile.save(using=self._db)
-
+    def create_employee_profile(self, user: User, name: str, email: str = None):
+        employee_profile = self.update_or_create_with_check(user=user, defaults={'name': name, 'email': email})
```
[the remark this answers is not in the document]
Comment (Feazom) [reply open_schools_platform/organization_management/employees/models.py:29]: It won't fail with same number **even if user is not deleted**

Usual get_or_update, create_or_update don't validate fields and may create wrong object

File: open_schools_platform/common/models.py [hunk 3]
```diff
@@ -1,17 +1,64 @@
...
+            obj = self.model(**params)
+
+        if isinstance(obj, AbstractBaseUser):
+            if password:
+                obj.set_password(password)
+            else:
+                obj.set_unusable_password()
+        obj.full_clean()
+        obj.save(using=self.db)
+        return obj
```
[the remark this answers is not in the document]
Comment (zavarin-michael) [reply open_schools_platform/common/models.py:58]: i think it would be better to redefine this methods in separate managers

**Review by zavarin-michael — APPROVED** [review]

**Pull request merged**